filebeat/module/kibana/log/ingest/pipeline.yml

```
@@ -37,18 +37,12 @@ processors:
     field: kibana.log.meta.res.statusCode
     target_field: http.response.status_code
     ignore_missing: true
-- rename:
-    field: kibana.log.meta.res.responseTime
-    target_field: temp.duration
-    ignore_missing: true
 - script:
     lang: painless
-    source: ctx.event.duration = Math.round(ctx.temp.duration * params.scale)
-    params:
-      scale: 1000000
-    if: ctx.temp?.duration != null
+    source: ctx.event.duration = Math.round(ctx.kibana.log.meta.res.responseTime * 1000000L)
```

Would this work if we continue using params?
```suggestion
    source: ctx.event.duration = Math.round(ctx.kibana.log.meta.res.responseTime * (long)params.scale)
```

---

I'll wait to see what @andrewkroh says above.